daddychukz/More-Recipes
# #151322263 Implement total recipe views

### What does this PR do?
It enables users see the number of times a recipe has been viewed

### Description of Task to be completed?

* Add 2 more fields to recipes table (viewCount and isViewed)
* Write function to take count of when a recipe is queried to be viewed
* Write function for user to change password
* Refactored codebase

### How should this be manually tested?

```
 1. From your terminal or console window, run 'npm run client'
 2. Navigate to 'http://localhost:8080/'
 3. Sign up with the App and Sign In
 4. Click on a Recipe to show details
 5. There you'll find the Views Icon with number of times opened
 6. Click on 'My Profile'
 7. Click on 'Change Password'
 8. Enter Old, New Password and confirm
 9. Click change password
 
```

Any background context you want to provide?

Nil

### What are the relevant pivotal tracker stories?

story type: feature
story Id: #151322263
story title: Users can see recipe views

### Screenshots (if appropriate)

<img width="1440" alt="screen shot 2018-01-24 at 6 36 03 pm" src="https://user-images.githubusercontent.com/30434186/35347551-c0f161c6-0135-11e8-91e9-d9bc78ab0aca.png">

<img width="1440" alt="screen shot 2018-01-24 at 6 36 17 pm" src="https://user-images.githubusercontent.com/30434186/35347552-c1145ab4-0135-11e8-9858-4f27e5a17604.png">
